subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -150,54 +132,80 @@ static void strncpy_nullterm(char *dst, const char *src, size_t maxlen)
 	}
 }
 
-int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
-		size_t payload_len, char *status)
+int aws_fota_parse_UpdateJobExecution_rsp(char *update_rsp_document,
+					  size_t payload_len, char *status)
 {
+	if (update_rsp_document == NULL || status == NULL) {
+		return -EINVAL;
+	}
+
 	struct update_rsp_obj rsp;
 
 	int ret = json_obj_parse(update_rsp_document, payload_len,
 			update_job_exec_stat_rsp_descr,
 			ARRAY_SIZE(update_job_exec_stat_rsp_descr), &rsp);
 
 	/* Check if the status field(1st field) of the object has been parsed */
-	if (ret & 0x01) {
+	if (ret & BIT(UPDATE_JOB_EXECUTION_STATUS_DECODED_BIT)) {
```

Comment:
You should check ret for errors before checking the bit, since the bit could be set in an retval. Same thing below.

Reply:
Yup added.